<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Index. taxize 0.0.3</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">

<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="#">taxize 0.0.3</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Index</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <div class="row">
  <div class="span8">
    <h1>This is <code>taxize</code></h1>

<p>We are developing <code>taxize</code> as a package to allow users to search over many websites for species names (scientific and common) and download up and downstream taxonomic hierarchical information - and many other things. </p>

<p>The <code>taxize</code> rOpenSci tutorial is <a href="https://github.com/ropensci/taxize_/wiki/taxize-tutorial">here</a></p>

<p><code>taxize</code> is part of the rOpenSci project, visit <a href="http://ropensci.org">our webiste</a> to learn more.</p>

<h3>Development by</h3>

<ul>
<li><a href="http://schamberlain.github.com/scott">Scott Chamberlain</a></li>
<li><a href="https://github.com/EDiLD">Eduard Sz√∂cs</a></li>
</ul>

<p>The functions in the package that hit a specific API have a prefix and suffix separated by an underscore. They follow the format of <code>service_whatitdoes</code>.  For example, <code>gnr_resolve</code> uses the Global Names Resolver API to resolve species names.  General functions in the package that don&#39;t hit a specific API don&#39;t have two words separated by an underscore, e.g., <code>classification</code>.</p>

<p>You need API keys for Encyclopedia of Life (EOL), the Universal Biological Indexer and Organizer (uBio), Tropicos, and Plantminer.</p>

<p>The following are URL&#39;s for API documentation, where to get API keys, and what prefix they have in function names. </p>

<h3>Currently implemented in <code>taxize</code></h3>

<ul>
<li>Encyclopedia of Life (EOL)

<ul>
<li><a href="http://www.eol.org/api/">API docs</a></li>
<li><a href="http://eol.org/users/register">Get an API key: start an account on EOL to get your API key</a></li>
<li><a href="https://eol.uservoice.com/forums/15429-encyclopedia-of-life-api">API forum</a></li>
<li>function prefix: <code>eol</code></li>
</ul></li>
<li>Taxonomic Name Resolution Service (TNRS) 

<ul>
<li><a href="http://tnrs.iplantcollaborative.org/api.html">API docs</a></li>
<li>NOTE!!!! Temporarily removed due to the API going down a lot. Will bring the function back when we build in better fail catching.</li>
</ul></li>
<li>Integrated Taxonomic Information Service (ITIS)

<ul>
<li><a href="http://www.itis.gov/ws_description.html">API docs</a></li>
<li>function prefix: <code>itis</code></li>
</ul></li>
<li>Phylomatic 

<ul>
<li><a href="http://www.phylodiversity.net/phylomatic/phylomatic_api.html">API docs</a></li>
<li>function prefix: <code>phylomatic</code></li>
</ul></li>
<li>uBio

<ul>
<li><a href="http://www.ubio.org/index.php?pagename=xml_services">API docs</a></li>
<li><a href="http://www.ubio.org/index.php?pagename=form">Get an API key</a></li>
<li>function prefix: <code>ubio</code></li>
</ul></li>
<li>Global Names Resolver (from EOL/GBIF)

<ul>
<li><a href="http://resolver.globalnames.org/">Use</a></li>
<li><a href="http://resolver.globalnames.org/api">API docs</a></li>
<li>function prefix: <code>gnr</code></li>
</ul></li>
<li>Global Names Index (from EOL/GBIF)

<ul>
<li><a href="http://gni.globalnames.org/">Use</a></li>
<li><a href="https://github.com/dimus/gni/wiki/api">API docs</a></li>
<li>function prefix: <code>gni</code></li>
</ul></li>
<li>IUCN Red List 

<ul>
<li><a href="https://www.assembla.com/spaces/sis/wiki/Red_List_API?version=3">API docs</a></li>
<li>function prefix: <code>iucn</code></li>
</ul></li>
<li>Tropicos (from Missouri Botanical Garden)

<ul>
<li><a href="http://services.tropicos.org/help">API docs</a></li>
<li><a href="http://services.tropicos.org/help?requestkey">Get an API key</a></li>
<li>function prefix: <code>tp</code></li>
</ul></li>
<li>Plantminer

<ul>
<li><a href="http://www.plantminer.com/">Their website</a></li>
<li><a href="http://www.plantminer.com/help">API docs</a></li>
<li>function prefix: <code>plantminer</code></li>
</ul></li>
<li>Theplantlist dot org

<ul>
<li><a href="http://www.theplantlist.org/">Their website</a></li>
<li>API docs: There are none! We wrap functions in the <a href="http://cran.r-project.org/web/packages/Taxonstand/index.html">taxonstand package</a></li>
<li>function prefix: <code>tpl</code></li>
</ul></li>
<li>Catalogue of Life

<ul>
<li><a href="http://www.catalogueoflife.org/colwebsite/content/services">API docs</a></li>
<li>function prefix: <code>col</code></li>
</ul></li>
<li>Global Invasive Species Database

<ul>
<li><a href="http://www.issg.org/database/welcome/">Their website</a></li>
<li>API docs: There are none! The function scraps the web directly.</li>
<li>function prefix: <code>gisd</code></li>
</ul></li>
<li>Freshwaterecology - The Taxa and Autecology Database for Freshwater Organisms

<ul>
<li><a href="http://www.freshwaterecology.info">Their website</a></li>
<li>API docs: There are none! The function scraps the web directly.</li>
<li>function prefix: <code>fresh</code></li>
<li>Note: Currently only the macro-invertebrate database is supported!</li>
</ul></li>
</ul>

<h3>Temporarily not implemented to resolve bugs or to complete development</h3>

<ul>
<li>Tree of Life web project

<ul>
<li><a href="http://tolweb.org/tree/phylogeny.html">Their website</a></li>
<li><a href="http://tolweb.org/tree/home.pages/downloadtree.html">API docs</a></li>
<li>function prefix: <code>tol</code></li>
</ul></li>
</ul>

<h3>Install <code>taxize</code></h3>

<ul>
<li>Stable version from CRAN:</li>
</ul>

<pre><code class="R">install.packages(&quot;taxize&quot;)
require(taxize)
</code></pre>

<ul>
<li>Or, development version from GitHub:</li>
</ul>

<pre><code class="R">install.packages(&quot;devtools&quot;)
require(devtools)
install_github(&quot;taxize_&quot;, &quot;ropensci&quot;)
require(taxize)
</code></pre>

<h3>Example hitting the TNRS (taxonomic names resolution service Phylotastic API):</h3>

<pre><code class="R">&gt; require(devtools)
&gt; install_github(&quot;taxize_&quot;,&quot;ropensci&quot;)
&gt; require(taxize)
&gt; mynames &lt;- c(&quot;Crepis atrabarba&quot;, &quot;Zygadenus venenosus&quot;)
&gt; mynames
[1] &quot;Crepis atrabarba&quot;    &quot;Zygadenus venenosus&quot;
&gt; tnrastic(query = mynames, output = &#39;names&#39;)
Your request is being processed. You can retrieve the results at http://api.phylotastic.org/tnrs/retrieve/c8b544f0794e13a61b0b63ea7952f664.
Pausing a bit for the query to finish...
      AcceptedName    sourceId MatchScore       submittedName
1 Crepis atribarba iPlant TNRS       0.98    Crepis atrabarba
2                  iPlant TNRS          1 Zygadenus venenosus
</code></pre>


    <h2>Help topics</h2>

    <h3></h3>
    
    
    <ul class="index">
        
      <li>
        <code><a href="capwords.html">capwords</a></code><br />Capitalize the first letter of a character string.</li>
            
      <li>
        <code><a href="classification.html">classification</a></code>(classification.tsn, classification.uid)<br />Get taxonomic hierarchy for a given taxon ID.</li>
            
      <li>
        <code><a href="col_children.html">col_children</a></code><br />Search Catalogue of Life for for direct children of a particular taxon.</li>
            
      <li>
        <code><a href="col_classification.html">col_classification</a></code><br />Search Catalogue of Life for taxonomic classifications.</li>
            
      <li>
        <code><a href="col_downstream.html">col_downstream</a></code><br />Use Catalogue of Life to get downstream taxa to a given taxonomic level.</li>
            
      <li>
        <code><a href="eol_dataobjects.html">eol_dataobjects</a></code><br />Given the identifier for a data object, return all metadata about the object.</li>
            
      <li>
        <code><a href="eol_hierarchy.html">eol_hierarchy</a></code><br />Retrieve all taxonomic hierarchy from given EOL taxonID.</li>
            
      <li>
        <code><a href="eol_pages.html">eol_pages</a></code><br />Search for pages in EOL database using a taxonconceptID.</li>
            
      <li>
        <code><a href="eol_ping.html">eol_ping</a></code><br />Ping EOL API to see if it's working.</li>
            
      <li>
        <code><a href="eol_search.html">eol_search</a></code><br />Search for terms in EOL database.</li>
            
      <li>
        <code><a href="fresh_codes.html">fresh_codes</a></code><br />Query species codes from </li>
            
      <li>
        <code><a href="fresh_desc.html">fresh_desc</a></code><br />Lookup-table for fresh_traits</li>
            
      <li>
        <code><a href="fresh_traits.html">fresh_traits</a></code><br />Query ecological parameters from </li>
            
      <li>
        <code><a href="fresh_validate.html">fresh_validate</a></code><br />Interface to the Taxa Validation Tool (TVT) of </li>
            
      <li>
        <code><a href="get_genes_avail.html">get_genes_avail</a></code><br />Retrieve gene sequences available for a species from NCBI.</li>
            
      <li>
        <code><a href="get_seqs.html">get_seqs</a></code><br />Retrieve nucleotide sequences from NCBI.</li>
            
      <li>
        <code><a href="get_tsn.html">get_tsn</a></code><br />Get the TSN code for a search term.</li>
            
      <li>
        <code><a href="get_uid.html">get_uid</a></code><br />Get the UID codes from NCBI for species names.</li>
            
      <li>
        <code><a href="gisd_isinvasive.html">gisd_isinvasive</a></code><br />Check invasive species status for a set of species from GISD database</li>
            
      <li>
        <code><a href="gni_details.html">gni_details</a></code><br />Search for taxonomic name details using the Global Names Index.</li>
            
      <li>
        <code><a href="gni_search.html">gni_search</a></code><br />Search for taxonomic names using the Global Names Index.</li>
            
      <li>
        <code><a href="gnr_datasources.html">gnr_datasources</a></code><br />Get data sources for the Global Names Resolver.</li>
            
      <li>
        <code><a href="gnr_resolve.html">gnr_resolve</a></code><br />Resolve names using Global Names Resolver.</li>
            
      <li>
        <code><a href="itis.html">itis</a></code><br />Use any API method for the ITIS API.</li>
            
      <li>
        <code><a href="itis_acceptname.html">itis_acceptname</a></code><br />Retrieve accepted TSN (with accepted name).</li>
            
      <li>
        <code><a href="itis_downstream.html">itis_downstream</a></code><br />Retrieve all taxa names or TSNs downstream in hierarchy from given TSN.</li>
            
      <li>
        <code><a href="itis_name.html">itis_name</a></code><br />Get taxonomic names for a given taxonomic name query.</li>
            
      <li>
        <code><a href="itis_phymat_format.html">itis_phymat_format</a></code><br />Get family names to make Phylomatic input object, and output input string
</li>
            
      <li>
        <code><a href="itis_taxrank.html">itis_taxrank</a></code><br />Retrieve taxonomic rank name from given TSN.</li>
            
      <li>
        <code><a href="iucn_summary.html">iucn_summary</a></code><br />Get a summary from the IUCN Red List.</li>
            
      <li>
        <code><a href="phylomatic_tree.html">phylomatic_tree</a></code><br />Format tree string, submit to Phylomatic, get newick tree.</li>
            
      <li>
        <code><a href="plantminer.html">plantminer</a></code><br />Search for taxonomy data from Plantminer.com</li>
            
      <li>
        <code><a href="rank_ref.html">rank_ref</a></code><br />Lookup-table for IDs of taxonomic ranks</li>
            
      <li>
        <code><a href="tax_name.html">tax_name</a></code><br />Get taxonomic names for a given taxonomic name query.</li>
            
      <li>
        <code><a href="taxize-package.html">taxize-package</a></code>(taxize)<br />Taxnomic search and phylogeny retrieval.</li>
            
      <li>
        <code><a href="tp_acceptednames.html">tp_acceptednames</a></code><br />Return all accepted names for a taxon name with a given id.</li>
            
      <li>
        <code><a href="tp_namedistributions.html">tp_namedistributions</a></code><br />Return all distribution records for for a taxon name with a given id.</li>
            
      <li>
        <code><a href="tp_namereferences.html">tp_namereferences</a></code><br />Return all reference records for for a taxon name with a given id.</li>
            
      <li>
        <code><a href="tp_summary.html">tp_summary</a></code><br />Return summary data a taxon name with a given id.</li>
            
      <li>
        <code><a href="tp_synonyms.html">tp_synonyms</a></code><br />Return all synonyms for a taxon name with a given id.</li>
            
      <li>
        <code><a href="tpl_get.html">tpl_get</a></code><br />Get The Plant List csv files.</li>
            
      <li>
        <code><a href="tpl_search.html">tpl_search</a></code><br />A light wrapper around the taxonstand fxn to call Theplantlist.org database.</li>
            
      <li>
        <code><a href="ubio_namebank.html">ubio_namebank</a></code><br />Search uBio namebank.</li>
    
    </ul>
  </div>
  
  <div class="span3 offset1">
        
    <h2>Dependencies</h2>
    <ul>
      <li><strong>Depends</strong>: </li>
      <li><strong>Imports</strong>: XML, RCurl, stringr, plyr, httr, RJSONIO, foreach, ape, Taxonstand, NCBI2R</li>
      <li><strong>Suggests</strong>: testthat, roxygen2, doMC, rjson</li>
      <li><strong>Extends</strong>: </li>
    </ul>
     
    <h2>Authors</h2>
    
    <h2>Authors</h2>
    
    <h2>Authors</h2>
    
    <h2>Authors</h2>
    
    
  </div>
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>
<p>Built by <a href="https://github.com/hadley/staticdocs">staticdocs</a>. Styled with <a href="http://twitter.github.com/bootstrap">bootstrap</a>.</p>
      </footer>
    </div>
  </body>
</html>