<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Index. taxize 0.0.3</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">

<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="navbar navbar-fixed-top">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="#">taxize 0.0.3</a>
    </div>
  </div>
</div>

<br><br><br><br>
    <div class="container">
      <div class="hero-unit">
        <h1>taxize 0.0.3</h1>
        <p>taxonomy in R</p>
      </div>
      
      <div class="row">
  <div class="span8">
    <h2>Help topics</h2>

    <h3></h3>
    
    
    <ul class="index">
        
      <li>
        <code><a href="capwords.html">capwords</a></code>: Capitalize the first letter of a character string.</li>
            
      <li>
        <code><a href="classification.html">classification</a></code>(classification.tsn, classification.uid): Get taxonomic hierarchy for a given taxon ID.</li>
            
      <li>
        <code><a href="col_children.html">col_children</a></code>: Search Catalogue of Life for for direct children of a particular taxon.</li>
            
      <li>
        <code><a href="col_classification.html">col_classification</a></code>: Search Catalogue of Life for taxonomic classifications.</li>
            
      <li>
        <code><a href="col_downstream.html">col_downstream</a></code>: Use Catalogue of Life to get downstream taxa to a given taxonomic level.</li>
            
      <li>
        <code><a href="eol_dataobjects.html">eol_dataobjects</a></code>: Given the identifier for a data object, return all metadata about the object.</li>
            
      <li>
        <code><a href="eol_hierarchy.html">eol_hierarchy</a></code>: Retrieve all taxonomic hierarchy from given EOL taxonID.</li>
            
      <li>
        <code><a href="eol_pages.html">eol_pages</a></code>: Search for pages in EOL database using a taxonconceptID.</li>
            
      <li>
        <code><a href="eol_ping.html">eol_ping</a></code>: Ping EOL API to see if it's working.</li>
            
      <li>
        <code><a href="eol_search.html">eol_search</a></code>: Search for terms in EOL database.</li>
            
      <li>
        <code><a href="fresh_codes.html">fresh_codes</a></code>: Query species codes from </li>
            
      <li>
        <code><a href="fresh_desc.html">fresh_desc</a></code>: Lookup-table for fresh_traits</li>
            
      <li>
        <code><a href="fresh_traits.html">fresh_traits</a></code>: Query ecological parameters from </li>
            
      <li>
        <code><a href="fresh_validate.html">fresh_validate</a></code>: Interface to the Taxa Validation Tool (TVT) of </li>
            
      <li>
        <code><a href="get_genes_avail.html">get_genes_avail</a></code>: Retrieve gene sequences available for a species from NCBI.</li>
            
      <li>
        <code><a href="get_seqs.html">get_seqs</a></code>: Retrieve nucleotide sequences from NCBI.</li>
            
      <li>
        <code><a href="get_tsn.html">get_tsn</a></code>: Get the TSN code for a search term.</li>
            
      <li>
        <code><a href="get_uid.html">get_uid</a></code>: Get the UID codes from NCBI for species names.</li>
            
      <li>
        <code><a href="gisd_isinvasive.html">gisd_isinvasive</a></code>: Check invasive species status for a set of species from GISD database</li>
            
      <li>
        <code><a href="gni_details.html">gni_details</a></code>: Search for taxonomic name details using the Global Names Index.</li>
            
      <li>
        <code><a href="gni_search.html">gni_search</a></code>: Search for taxonomic names using the Global Names Index.</li>
            
      <li>
        <code><a href="gnr_datasources.html">gnr_datasources</a></code>: Get data sources for the Global Names Resolver.</li>
            
      <li>
        <code><a href="gnr_resolve.html">gnr_resolve</a></code>: Resolve names using Global Names Resolver.</li>
            
      <li>
        <code><a href="itis.html">itis</a></code>: Use any API method for the ITIS API.</li>
            
      <li>
        <code><a href="itis_acceptname.html">itis_acceptname</a></code>: Retrieve accepted TSN (with accepted name).</li>
            
      <li>
        <code><a href="itis_downstream.html">itis_downstream</a></code>: Retrieve all taxa names or TSNs downstream in hierarchy from given TSN.</li>
            
      <li>
        <code><a href="itis_name.html">itis_name</a></code>: Get taxonomic names for a given taxonomic name query.</li>
            
      <li>
        <code><a href="itis_phymat_format.html">itis_phymat_format</a></code>: Get family names to make Phylomatic input object, and output input string
</li>
            
      <li>
        <code><a href="itis_taxrank.html">itis_taxrank</a></code>: Retrieve taxonomic rank name from given TSN.</li>
            
      <li>
        <code><a href="iucn_summary.html">iucn_summary</a></code>: Get a summary from the IUCN Red List.</li>
            
      <li>
        <code><a href="phylomatic_tree.html">phylomatic_tree</a></code>: Format tree string, submit to Phylomatic, get newick tree.</li>
            
      <li>
        <code><a href="plantminer.html">plantminer</a></code>: Search for taxonomy data from Plantminer.com</li>
            
      <li>
        <code><a href="rank_ref.html">rank_ref</a></code>: Lookup-table for IDs of taxonomic ranks</li>
            
      <li>
        <code><a href="tax_name.html">tax_name</a></code>: Get taxonomic names for a given taxonomic name query.</li>
            
      <li>
        <code><a href="taxize-package.html">taxize-package</a></code>(taxize): Taxnomic search and phylogeny retrieval.</li>
            
      <li>
        <code><a href="tp_acceptednames.html">tp_acceptednames</a></code>: Return all accepted names for a taxon name with a given id.</li>
            
      <li>
        <code><a href="tp_namedistributions.html">tp_namedistributions</a></code>: Return all distribution records for for a taxon name with a given id.</li>
            
      <li>
        <code><a href="tp_namereferences.html">tp_namereferences</a></code>: Return all reference records for for a taxon name with a given id.</li>
            
      <li>
        <code><a href="tp_summary.html">tp_summary</a></code>: Return summary data a taxon name with a given id.</li>
            
      <li>
        <code><a href="tp_synonyms.html">tp_synonyms</a></code>: Return all synonyms for a taxon name with a given id.</li>
            
      <li>
        <code><a href="tpl_get.html">tpl_get</a></code>: Get The Plant List csv files.</li>
            
      <li>
        <code><a href="tpl_search.html">tpl_search</a></code>: A light wrapper around the taxonstand fxn to call Theplantlist.org database.</li>
            
      <li>
        <code><a href="ubio_namebank.html">ubio_namebank</a></code>: Search uBio namebank.</li>
    
    </ul>
  </div>
  
  <div class="span3 offset1">
        
    <h2>Dependencies</h2>
    <ul>
      <li><strong>Depends</strong>: </li>
      <li><strong>Imports</strong>: XML, RCurl, stringr, plyr, httr, RJSONIO, foreach, ape, Taxonstand, NCBI2R</li>
      <li><strong>Suggests</strong>: testthat, roxygen2, doMC, rjson</li>
    </ul>
    
    <h2>Authors</h2>
          Scott Chamberlain, Eduard Sz√∂cs, Carl Boettiger, and Ignasi Bartomeus
    
  </div>
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>
<p>Built by <a href="https://github.com/hadley/staticdocs">staticdocs</a>. Styled with <a href="http://twitter.github.com/bootstrap">bootstrap</a>.</p>
      </footer>
    </div>
  </body>
</html>