<!--
%\VignetteEngine{knitr}
%\VignetteIndexEntry{An R Markdown Vignette made with knitr}
-->

taxize vignette - a taxonomic toolbelt for R
======

### About the package

`taxize` is a taxonomic toolbelt for R. `taxize` wraps APIs for a large suite of taxonomic databases availab on the web.

********************

### Quick start

#### First, install `taxize`

First, install and load `taxize` into the R session.

```{r installtaxizecran, eval=FALSE}
install.packages("taxize")
library(taxize)
```

Advanced users can also download and install the latest development copy from [GitHub](https://github.com/ropensci/taxize_).

#### Resolve taxonomic name

This is a common task in biology. We often have a list of species names and we want to know a) if we have the most up to date names, b) if our names are spelled correctly, and c) the scientific name for a common name. One way to resolve names is via the Global Names Resolver (GNR) service provided by the [Encyclopedia of Life][eol]. Here, we are searching for two misspelled names:

```{r resolvenames, eval=TRUE, cache=TRUE}
temp <- gnr_resolve(names = c("Helianthos annus", "Homo saapiens"))
temp[ , -c(1,4)]
```

The correct spellings are *Helianthus annuus* and *Homo sapiens*. Another approach uses the [Taxonomic Name Resolution Service via the Taxosaurus API][taxosaurus] developed by iPLant and the Phylotastic organization. In this example, we provide a list of species names, some of which are misspelled, and we'll call the API with the *tnrs* function.

```{r tnrs, eval=TRUE, cache=TRUE}
mynames <- c("Helianthus annuus", "Pinus contort", "Poa anua", "Abis magnifica",
  	"Rosa california", "Festuca arundinace", "Sorbus occidentalos","Madia sateva")
tnrs(query = mynames)[ , -c(5:7)]
```

It turns out there are a few corrections: e.g., *Madia sateva* should be *Madia sativa*, and *Rosa california* should be *Rosa californica*. Note that this search worked because fuzzy matching was employed to retrieve names that were close, but not exact matches. Fuzzy matching is only available for plants in the TNRS service, so we advise using EOL's Global Names Resolver if you need to resolve animal names.

taxize takes the approach that the user should be able to make decisions about what resource to trust, rather than making the decision. Both the EOL GNR and the TNRS services provide data from a variety of data sources. The user may trust a specific data source, thus may want to use the names from that data source. In the future, we may provide the ability for taxize to suggest the best match from a variety of sources.

Another common use case is when there are many synonyms for a species. In this example, we have three synonyms of the currently accepted name for a species. 

```{r synonyms, eval=TRUE, cache=TRUE, message=FALSE, warning=FALSE, comment=NA}
mynames <- c("Helianthus annuus ssp. jaegeri", "Helianthus annuus ssp. lenticularis", "Helianthus annuus ssp. texanus", "Helianthus annuus var. lenticularis", "Helianthus annuus var. macrocarpus", "Helianthus annuus var. texanus")
tsn <- get_tsn(mynames)
ldply(tsn, itis_acceptname)
```

#### Retrieve higher taxonomic names

Another task biologists often face is getting higher taxonomic names for a taxa list. Having the higher taxonomy allows you to put into context the relationships of your species list. For example, you may find out that species A and species B are in Family C, which may lead to some interesting insight, as opposed to not knowing that Species A and B are closely related. This also makes it easy to aggregate/standardize data to a specific taxonomic level (e.g., family level) or to match data to other databases with different taxonomic resolution (e.g., trait databases).

A number of data sources in taxize provide the capability to retrieve higher taxonomic names, but we will highlight two of the more useful ones: [Integrated Taxonomic Information System (ITIS)][itis] and [National Center for Biotechnology Information (NCBI)][ncbi]. First, we'll search for two species, *Abies procera} and *Pinus contorta* within ITIS.

```{r itisclass, eval=TRUE, cache=TRUE}
specieslist <- c("Abies procera","Pinus contorta")
classification(specieslist, db = 'itis')
```

It turns out both species are in the family Pinaceae. You can also get this type of information from the NCBI by doing `classification(specieslist, db = 'ncbi')`.

Instead of a full classification, you may only want a single name, say a family name for your species of interest. The function *tax_name} is built just for this purpose. As with the `classification` function you can specify the data source with the `db` argument, either ITIS or NCBI. 

```{r taxname, eval=TRUE,cache=TRUE}
tax_name(query = "Helianthus annuus", get = "family", db = "itis")
tax_name(query = "Helianthus annuus", get = "family", db = "ncbi")
```

I may happen that a data source does not provide information on the queried species, than one could take the result from another source and union the results from the different sources.

#### Interactive name selection
As mentioned most databases use a numeric code to reference a species. A general workflow in taxize is: Retrieve Code for the queried species and then use this code to query more data/information.

Below are a few examples. When you run these examples in R, you are presented with a command prompt asking for the row that contains the name you would like back; that output is not printed below for brevity. In this example, the search term has many matches. The function returns a data.frame of the matches, and asks for the user to input what row number to accept. 

```{r interactive, eval=TRUE, echo=TRUE, cache=TRUE}
get_tsn(searchterm = "Heliastes", searchtype = "sciname")
```

In another example, you can pass in a long character vector of taxonomic names:

```{r interactive_many, eval=TRUE, echo=TRUE, cache=TRUE}
splist <- c("annona cherimola", 'annona muricata', "quercus robur",
  	"shorea robusta", "pandanus patina", "oryza sativa", "durio zibethinus")
get_tsn(searchterm = splist, searchtype = "sciname")
```

In another example, note that no match at all returns an NA:

```{r interactive_na, eval=TRUE, echo=TRUE, cache=TRUE}
get_uid(sciname = c("Chironomus riparius", "aaa vva"))
```

#### Retrieve a phylogeny

Ecologists are increasingly taking a phylogenetic approach to ecology, applying phylogenies to topics such as the study of community structure, ecological networks, functional trait ecology. Yet, Many biologists are not adequately trained in reconstructing phylogenies. Fortunately, there are some sources for getting a phylogeny without having to know how to build one; one of these is for angiosperms, called Phylomatic \cite{webb2005}. We have created a workflow in taxize that accepts a species list, and taxize works behind the scenes to get higher taxonomic names, which are required by Phylomatic to get a phylogeny. Here is a short example, producing the tree in figure (Fig. 1).

```{r phylomaticphylogeny_disp, eval=FALSE}
taxa <- c("Poa annua", "Abies procera", "Helianthus annuus")
tree <- phylomatic_tree(taxa = taxa)
tree$tip.label <- capwords(tree$tip.label)
plot(tree, cex=1)
```

Input the taxonomic names

```{r phylomaticphylogeny1, eval=TRUE, dev=c ('pdf','tiff'), dpi=300, out.width='4in', out.height='4in', cache=TRUE, include=FALSE}
taxa <- c("Poa annua", "Abies procera", "Helianthus annuus")
```

Fetch the tree - the formatting of names and higher taxonmy is done within the function

```{r phylomaticphylogeny2, eval=TRUE, dev=c ('pdf','tiff'), dpi=300, out.width='4in', out.height='4in', cache=TRUE, include=FALSE}
tree <- phylomatic_tree(taxa = taxa)
```

Captilize the species names

```{r phylomaticphylogeny3, eval=TRUE, dev=c ('pdf','tiff'), dpi=300, out.width='4in', out.height='4in', cache=TRUE, include=FALSE}
tree$tip.label <- capwords(tree$tip.label)
```

Plot the tree

```{r phylomaticphylogeny4, eval=TRUE, dev=c ('pdf','tiff'), dpi=300, out.width='4in', out.height='4in', cache=TRUE, include=FALSE}
plot(tree, cex=1)
```

Behind the scenes the function *phylomatic_tree} retrieves a Taxonomic Serial Number (TSN) from ITIS for each species name, then a string is created for each species like this `poaceae/oryza/oryza_sativa` (with format `family/genus/genus_epithet`). These strings are submitted to the Phylomatic API, and if no errors occur, a phylogeny in newick format is returned. The `phylomatic_tree()` function also cleans up the newick string and converts it to an **ape** `phylo` object. The output from `phylomatic_tree()` is a `phylo` object, which can be used for plotting and phylogenetic analyses. Be aware that Phylomatic has certain limitations - refer to the paper describing [Phylomatic][phylomatic] and the website [http://phylodiversity.net/phylomatic/][http://phylodiversity.net/phylomatic/].

There are currently no resources for getting a phylogeny of animals simply from species names. However, a few projects are working on this problem, including the [Open Tree of Life][opentree]. We will incorporate these resources when the appropriate APIs are available.


#### What taxa are the children of my taxon of interest?

If someone is not a taxonomic specialist on a particular taxon he likely does not know what children taxa are within a family, or within a genus. This task becomes especially unwieldy when there are a large number of taxa downstream. You can of course go to a website like [Wikispecies][wikispecies] or [Encyclopedia of Life][eol] to get downstream names. However, taxize provides an easy way to programatically search for downstream taxa, both for the [Catalogue of Life (CoL)][col] and the [Integrated Taxonomic Information System][itis]. Here is a short example using the CoL in which we want to find all the species within the genus *Apis* (honey bees).

```{r downstream, eval=TRUE, cache=TRUE}
col_downstream(name = "Apis", downto = "Species")[[1]]
```

The result from the above call to `col_downstream()` is a data.frame that gives a number of columns of different information. 

#### IUCN Status

There are a number of things we can do once we have the correct taxonomic names. One thing we can do is ask about the conservation status of a species ([IUCN Red List of Threatened Species][iucn]). We have provided a set of functions, `iucn_summary` and `iucn_status`, to search for species names, and extract the status information, respectively. Here, we search for the Panther and *Lynx*.

```{r iucn, eval=TRUE, cache=TRUE}
ia <- iucn_summary(c("Panthera uncia", "Lynx lynx"))
iucn_status(ia)
```

It turns out that the panther has a status of endangered (EN) and the lynx has a status of least concern (LC).

#### Search for available genes in GenBank

Another use case available in taxize deals with genetic sequences. taxize has three functions to interact with GenBank to search for available genes (`get_genes_avail`), download genes by GenBank ID (`get_genes`), and download genes via taxonomic name search, including retrieving a congeneric if the searched taxon does not exist in the database (`get_seqs`). In this example, we search for gene sequences for *Umbra limi*.

```{r genbank_one, eval=TRUE, cache=TRUE}
out <- get_genes_avail(taxon_name = "Umbra limi", seqrange = "1:2000", getrelated = FALSE)
```

Then we can ask if *RAG1* exists in any of the gene names.

```{r genbank_two, eval=TRUE, cache=TRUE}
out[grep("RAG1", out$genesavail, ignore.case = TRUE),]
```

It turns out that there are 430 different unique records found. However, this doesn't mean that there are 430 different genes found as the API does not provide metadata to classify genes. However, you can use regular expressions (e.g., `grep`) to search for the gene of interest.


#### Matching species tables with different taxonomic resolution

Biologist often need to match different sets of data tied to species. For example, trait-based approaches are a promising tool in ecology. One problem is that abundance data must be matched with trait databases. These two data tables may contain species information on different taxonomic levels and possibly data must be aggregated to a joint taxonomic level, so that the data can be merged. taxize can help in this data-cleaning step, providing a reproducible workflow: 

We can use the mentioned `classification`-function to retrieve the taxonomic hierarchy and then search the hierarchies up- and downwards for matches. Here is an example to match a species with names on three different taxonomic levels.

```{r tax_match, cache=TRUE, tidy=FALSE}
A <- "gammarus roeseli" 

B1 <- "gammarus roeseli"
B2 <- "gammarus"
B3 <- "gammaridae"

A_clas <- classification(A, db = 'ncbi')
B1_clas <- classification(B1, db = 'ncbi')
B2_clas <- classification(B2, db = 'ncbi')
B3_clas <- classification(B3, db = 'ncbi')


B1[match(A, B1)]
A_clas[[1]]$Rank[tolower(A_clas[[1]]$ScientificName) %in% B2]
A_clas[[1]]$Rank[tolower(A_clas[[1]]$ScientificName) %in% B3]
```

If we find a direct match (here *Gammarus roeseli*), we are lucky. But we can also match Gammaridae with *Gammarus roeseli*, but on a lower taxonomic level. A more comprehensive and realistic example (matching a trait table with an abundance table) is given in the vignette on matching.


#### Aggregating data to a specific taxonomic rank

In biology, one can asks questions at varying taxonomic levels. One may perform analyses on different taxonomic levels. This use case is easily handled in taxize. A function called `tax_agg` will aggregate community data to a specific taxonomic level. In this example, we take data of 5 species and aggregate them to family level. Again we can specify if we want to use data from ITIS or NCBI.

```{r tax_agg, eval=TRUE, cache=TRUE, tidy=FALSE}
data(dune, package = 'vegan')
df <- dune[ , 1:5]
colnames(df) <- c("Bellis perennis", "Empetrum nigrum", "Juncus bufonius", 
                  "Juncus articulatus", "xxx")
head(df)
agg <- tax_agg(df, rank = 'family', db = 'ncbi')
agg
head(agg$x)
```

[eol]: 
[taxosaurus]: 
[ncbi]: 
[itis]: 
[phylomatic]: 
[opentree]: 
[wikispecies]: 
[col]: 
[iucn]: 